src/hooks/useVerifyDate.ts

```diff
@@ -0,0 +1,41 @@
+export const useVerifyDate = () => {
+  const formatDateLocalString = (inputDateString: string) => {
+    try {
+      const dateObject = new Date(inputDateString);
+      const formattedDay = dateObject.getUTCDate().toString().padStart(2, "0");
+      const formattedMonth = (dateObject.getUTCMonth() + 1).toString().padStart(2, "0");
+      const formattedYear = dateObject.getUTCFullYear();
+      const formattedDate = `${formattedDay}/${formattedMonth}/${formattedYear}`;
+      return formattedDate;
+    } catch (error) {
+      return "Invalid Date";
+    }
+  };
+
+  const formatDateString = (inputDateString: string) => {
+    if (!/^\d{4}-\d{2}-\d{2}$/.test(inputDateString)) {
+      return "Invalid Date";
+    }
+
+    const [year, month, day] = inputDateString.split("-");
+    const dateObject = new Date(Number(year), Number(month) - 1, Number(day));
+
+    const formattedDay = dateObject.getDate().toString().padStart(2, "0");
+    const formattedMonth = (dateObject.getMonth() + 1).toString().padStart(2, "0");
+    const formattedYear = dateObject.getFullYear();
+
+    const formattedDate = `${formattedDay}/${formattedMonth}/${formattedYear}`;
+
+    return formattedDate;
+  };
+
+  function isDateStringValid(dateString: string): boolean {
+    if (!isNaN(Number(dateString))) {
+      return false;
+    }
+    const timestamp = Date.parse(dateString);
+    return !isNaN(timestamp);
+  }
+
+  return { formatDateLocalString, formatDateString, isDateStringValid };
```

Comment:
The code looks good to me aside from one thing: have we considered bringing in a library to handle date parsing and display? Luxon for instance is very powerful, and date parsing/display is fraught enough that it's really easy to miss edge cases and end up with a buggy front end.

Reply:
Oh, and as I read this a bit closer; I don't see any reason for this to be a hook. It's not providing any contextual information for a component, and would be better done as a simply utility file with these three functions exported (unless we add a date math library like Luxon, in which case this is likely superfluous)